evaluator.js

```diff
@@ -155,7 +153,7 @@ function analyze_variable_declaration(stmt) {
     return (env, succeed, fail) => {
         value_func(env,
                    (value, fail2) => {
-                       set_name_value(name, value, env);
+                       set_name_value(name, value, env, true);
```

Comment:
The `true` argument is necessary for `set_name_value` as it indicates that the name being declared is a variable (and not a constant).